File: blormlib/src/main/java/br/com/bloder/blormlib/validation/validations/Filled.java
package br.com.bloder.blormlib.validation.validations;

import android.widget.EditText;

import br.com.bloder.blormlib.validation.Validate;
import br.com.bloder.blormlib.validation.Validation;

/**
 * Created by bloder on 05/06/16.
 */
public class Filled extends Validation {

  public EditText editText;
  private int maxSize = -1;
  private int minSize = -1;
 
  public Filled withMaxSize(int maxSize) {
    this.maxSize = maxSize;
    return this;
  }
 
  public Filled withMinimumSize(int minSize) {
    this.minSize = minSize;
    return this;
  }
 
  @Override public Validate validate() {
    return new Validate() {
      @Override public boolean validate() {
        editText = (EditText) field;
        String text = String.valueOf(editText.getText()).trim();
        int textLength = text.length();
        if (!text.isEmpty()) {
          if (maxSize != -1) {
            if (textLength > maxSize) {
              return false;
            }
          }
          if (minSize != -1) {
            if (textLength < minSize) {
              return false;
            }
          }
          return true;
        }
        return false;
      }

      @Override
      public void onError() {
        editText.setError(errorMessage != null && !errorMessage.isEmpty() ? errorMessage : "This field can't be blank");
      }

      @Override public void onSuccess() {
        editText.setError(null);
      }
    };
  }
}

File: blormlib/src/main/java/br/com/bloder/blormlib/validation/validations/Filled.kt
package br.com.bloder.blormlib.validation.validations

import android.widget.EditText
import br.com.bloder.blormlib.validation.Validate
import br.com.bloder.blormlib.validation.Validation

/**
 * Created by bloder on 03/09/16.
 */
class Filled : Validation(), Validate {

    private var editText: EditText? = null
    var maxSize: Int = -1
    var minSize: Int = -1

    fun withMinimunSize(minSize: Int) : Filled {
        this.minSize = minSize
        return this
    }
    fun withMaxSize(maxSize: Int) : Filled {
        this.maxSize = maxSize
        return this
    }

    override fun validate(): Validate = this
    override fun validateIt(): Boolean {
        editText = field as EditText
        val text: String = editText?.text.toString()
        val textLength: Int = text.length
        if (minSize != -1 && maxSize != -1) {
            return !text.isEmpty() && textLength >= minSize || !text.isEmpty() && textLength <= maxSize
        } else {
            return text != ""
        }
    }
    override fun onError() {
        editText?.error = errorMessage ?: "This field can't be blank"
    }
    override fun onSuccess() {
        editText?.error = null
    }
}